industry_benchmarks/utils/data_gathering.py

```diff
@@ -0,0 +1,511 @@
+        ligand_scores = {}
+        num_rotatable_bonds = get_number_rotatable_bonds(node)
+        ligand_scores["num_rotatable_bonds"] = num_rotatable_bonds
+        num_rings = get_number_ring_systems(node)
+        ligand_scores["num_rings"] = num_rings
+        num_heavy_atoms = get_number_heavy_atoms(node)
+        ligand_scores["num_heavy_atoms"] = num_heavy_atoms
+        num_elements = get_system_element_count(node)
+        ligand_scores["num_elements"] = num_elements
+        sasa = get_solvent_accessible_surface_area(node)
+        ligand_scores["solvent_accessible_surface_area"] = sasa
+
+        all_ligand_scores[name] = ligand_scores
+
+    return all_ligand_scores
+
+
+@click.command
+@click.option(
+    '--input_ligand_network',
+    type=click.Path(dir_okay=False, file_okay=True, path_type=pathlib.Path),
+    default=pathlib.Path("./alchemicalNetwork/ligand_network.graphml"),
+    required=True,
+    help=("Path to the ligand_network.graphml file that was used to run these "
+         "simulations."),
+)
+@click.option(
+    '--output_dir',
+    type=click.Path(dir_okay=True, file_okay=False, path_type=pathlib.Path),
+    default=pathlib.Path("./"),
+    required=True,
+    help="Path to the output directory that stores all data.",
+)
+@click.option(
+    '--fixed_ligand_network',
+    type=click.Path(dir_okay=False, file_okay=True, path_type=pathlib.Path),
+    default=pathlib.Path("./ligand_network.graphml"),
+    required=False,
+    help=("Only needed when a broken network was fixed with additional edges. "
+          "Path to the ligand_network.graphml file that was used to run the "
+         "simulations of fixing the network."),
+)
+def gather_data(
+    input_ligand_network: pathlib.Path,
+    output_dir: pathlib.Path,
+    fixed_ligand_network: str =None,
+):
+    """
+    Function that gathers all the data.
+    """
+    # Make folder for outputs
+    output_dir.mkdir(exist_ok=False, parents=True)
+
+    # GATHER Input based information
+    ligand_network = parse_ligand_network(input_ligand_network)
+    # Combine old + new LigandNetwork if a fixed network is provided
+    if fixed_ligand_network:
+        fixed_network = parse_ligand_network(fixed_ligand_network)
+        ligand_network = ligand_network.enlarge_graph(
+            edges=fixed_network.edges, nodes=fixed_network.nodes)
+    transformation_scores = gather_transformation_scores(ligand_network)
+    ligand_scores = gather_ligand_scores(ligand_network)
+    # Create a single dict of all scores
+    scores = {
+        "transformation_scores": transformation_scores,
+        "ligand_scores": ligand_scores,
+    }
+    # Save this to json
+    file = pathlib.Path(output_dir / 'all_network_properties.json')
+    with open(file, mode='w') as f:
+        json.dump(scores, f)
+
+    blinded_network = get_transformation_network_map(ligand_network)
+    # Save this to json
+    file = pathlib.Path(output_dir / 'network_map.json')
```

Comment:
For the sake of our partners we will likely need to provide a single JSON with all this information (plus the free energy scores and other things) with a pretty formatting on.

This is an annoying detail but it's so that we can make legal review easy (and limit the amount of things they send us).

Reply:
Collapsed things into a single json file in commit [433bea4], though it may not be pretty yet.